### DISCUSSION TITLE
Fat FS for Teensy 4.1

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/iMXRT1062#24] [General]
This may be answered somewhere else. If so, I apologize .

Anyway, could someone point me to a repository that works on a Teensy 4.1 and grblHAL.

**Comment by terjeio** [comment]
Is the [uSDFS library](https://github.com/grblHAL/iMXRT1062#sd-card-plugin) broken again? Even after adding the patch I have provided?

**Reply by phil-barrett** [reply]
I'm using uSDFS (and USE_MSC is set to 0).  But get a build error when I enable SDCARD.  

> C:\Users\philba\AppData\Local\Temp\arduino_build_257797\libraries\grblHAL_Teensy4\networking\vfs.c.o: In function `vfs_getcwd':
> D:\Documents\Arduino\libraries\grblHAL_Teensy4\src\networking/vfs.c:150: undefined reference to `f_getcwd'
> collect2.exe: error: ld returned 1 exit status

I do not get this error when building with SDCARD_ENABLE commented out (but with ETHERNET_ENABLE).  

Note, I'm not sure why the quote displays weirdly.

EDIT: this is in sd_config.h
`#define USE_MSC 0	// will be used in sd_msc.cpp`

**Reply by terjeio** [reply]
This? "In addition to this ffconf.h has to be edited, #define FF_FS_RPATH value has to be changed to 2 (from 1) or you will get a compiler error."

**Reply by phil-barrett** [reply]
Ah, sorry, missed that.  thanks.

**Reply by terjeio** [reply]
There is another "bug" in the websocket code in the latest version - from a compiler issue? Compiles ok with the STM32CubeIDE but the Arduino compiler complains about a constant not beeing a constant...
